doc/adrs/XXX_support-server-side-pagination-for-table.adoc

```diff
@@ -0,0 +1,146 @@
+This approach will allow us to fetch and display data more efficiently, especially for large datasets.
+
+=== Cursor-Based vs. Page-Based Pagination
+
+When implementing server-side pagination, there are two primary approaches to consider: cursor-based and page-based pagination.
+
+==== Page-Based Pagination
+
+===== Advantages
+* Simplicity: Page-based pagination is simpler to implement and understand.
+* User Experience: It allows users to navigate directly to specific pages.
+
+===== Disadvantages
+* Performance: For large datasets, counting the total number of rows can be resource-intensive and slow down the response time.
+
+==== Cursor-Based Pagination
+
+===== Advantages
+* Efficiency: Cursor-based pagination is generally more efficient for large datasets as it avoids the need to count the total number of rows.
+* Consistency: It ensures consistency in the results, especially when the dataset is frequently updated.
+ The cursor acts as a pointer to the last fetched item, allowing for more accurate pagination.
+* Scalability: This approach scales better with large datasets and high-frequency updates.
+
+===== Disadvantages
+* Complexity: cursor-based pagination can be more complex to implement.
+* User Experience: It may not provide a straightforward way to jump to a specific page.
+* We cannot manage multiple `rowDescription` instances because we are unable to determine the number of rows returned for each description.
+
+==== Conclusion
+
+Performance considerations led us to choose the *cursor-based* pagination approach.
+The constraint of having only one instance of `rowDescription` is not inherently problematic.
+
+=== Material React Table
+
+To enable server-side pagination, we will set `manualPagination: true` in Material React Table.
+This will allow us to take control of pagination on the backend, ensuring that we can efficiently manage large datasets and accurately display the true total number of rows.
+The pagination must be managed within our own state, and listeners must be added to handle pagination state changes, allowing the backend to be called with the updated pagination data.
+
+[source, typescript]
+----
+const [pagination, setPagination] = useState({
+  pageIndex,
+  pageSize,
+});
+
+useEffect(() => {
+  //do something when the pagination state changes
+}, [pagination.pageIndex, pagination.pageSize]);
+
+const table = useMaterialReactTable({
+  onPaginationChange: setPagination,
+  state: { pagination },
+});
+
+
+----
+
+=== Frontend
+
+The pagination data will be added to the `representationId` of the subscription:
+
+[source, typescript]
+----
+`${tableId}?cursor=${cursor}&pageSize=${pageSize}`
+----
+
+=== GraphQL
+
+* Add pagination data in `Table` type
+
+[source, graphql]
+----
+type Table {
+
+  pagination: Pagination
+}
+
+type Pagination {
+  hasPreviousPage: Boolean!
+  hasNextPage: Boolean!
+  rowCount: Int
```

Comment:
Could you use `totalRowCount` to clarify that we are not talking about the number of rows actually returned by the query. Same comment on the line below